### ISSUE TITLE
Invalid plugin config file MyStaticlib.gdip while exporting ios project from Godot

### ISSUE THREAD

**Issue by kyadalu1** [LettucePie/ios-plugin-integrate-demo#3]
Extermely Sorry to open this as an issue, but i am trying to create my own ios plugin and need your help 

I am trying to learn how to create iOS plugin for Godot engine. I just created simple iOS plugin which stores some key value data in `UserDefaults` and retrieve it. Here are the steps which i followed

1. Download Godot source code from master branch
2. `brew install scons`
3. cd to Godot source code directory in terminal and ran `scons platform=ios target=template_release -j4 `
4. Open xcode and selected static library for iOS
5. Go to the downloaded godot-repo, navigate to platform / iOS folder and add this path to `Build settings` => `Header Search Paths` in Xcode
6. Add the required code to my objective-c file & selected my iOS simulator and ran Command + B for the library to be build which is successfull
7. Create my Godot poject in Godot editor and created a bin folder
8. Found the `libmystatticlib.a` from my iOS project and copy pasted it to the bin folder of my iOS project
9. Create a folder called iOS and inside that a plugins folder. In pulgins folder created `.gdip` file and copied the code from [docs](https://docs.godotengine.org/en/stable/tutorials/platform/ios/ios_plugin.html#creating-an-ios-plugin) and refactored it according to my needs
10. In Godot, click project => Export => Add iOS => Click Fix Import Error => Added bundle identifier & App Store team id
11. Created a button in Godot tscn and added pressed listener to gdscript file
12. On button pressed code , added the followed code

```
var bridge = load("res://ios/plugins/MyStaticlib.gdip").new()
bridge._save_value("Hello", "keyhello")
```

When i try to export my godot project for iOS i keep getting error saying `Invalid plugin config file MyStaticlib.gdip`

Which step am i doing wrong. Please help

[Here](https://github.com/kyadalu1/IosSamplePlugin) is the sample ios plugin repo
[Here](https://github.com/kyadalu1/GodotSampleApp) is the sample godot projectn repo

**Comment by LettucePie** [comment]
Hello. No need to feel sorry!

I no longer have my macbook or iphone, as I abandoned apple development. So I won't be able to reproduce your steps and witness the errors myself.

I personally experienced the Invalid Plugin Config File issue, and shared my solution to it in the Readme on this repo. It's located at the very bottom, third paragraph up. 

> If you encounter "Invalid Plugin Config File plugin.gdip", then you too now share the confusing experience that consumed a large part of my day. Initially when I followed along the official instructions [Here](https://github.com/godotengine/godot-ios-plugins#instructions) I encountered a lot of issues, and had great difficulty finding answers on this error code. Eventually I tried using the Released 3.5 plugins, and the error went away, but then I couldn't export to XCODE with the plugin enabled. Something about it failing to run Ld /blah/blah/blah/blah. What finally worked was the instructions I have provided you now. I at first assumed I would only need the template_debug and release_debug targets. It wasn't until I went ahead and just compiled all the different targets out of desperation that it finally worked.

So I suppose I will have no choice but to suggest you do the same, and recompile your plugin for all three targets.

- debug
- release
- release_debug

Unfortunately, since I don't have any access to xcode anymore (and setting up the linux trick would be difficult atm), this is the only advice I can provide. It looks as though you used your own scripts/methods of compiling the plugin libraries, which already flies over my level of experience on the matter. I only ever used the scripts provided by the contributors of the godot-ios-plugin repo (https://github.com/godotengine/godot-ios-plugins).

I wish you best of luck in this pursuit. It seems as things have gotten further out of hand over there, since even now they still haven't updated their release tags with 4.X compatible plugins...

**Comment by kyadalu1** [comment]
Thank you so much for your reply. I did figure it our for android and posted a video about it [here](https://www.youtube.com/watch?v=rn7iB4XQDmQ) & [here](https://www.youtube.com/watch?v=NWxA8Dx_6eo). Once i resolve it for ios i will do it for iOS as well
